### DISCUSSION TITLE
aggregate and resample_by_interpolation

### DISCUSSION THREAD

**Discussion by paulf81** [NatLabRockies/floris#931] [General]
the WindRose class has two functions for something like resampling, `aggregate` and `resample_by_interpolation`.  `aggregate` sums the frequencies together and only allows combining into fewer bins.  `resample_by_interpolation` uses interpolation of frequencies and can be used for upsampling or downsampling.  In my testing using `resample_by_interpolation` for upsampling works in an intuitive way but for downsampling can work, I think basically correctly, but in perhaps surprising way, since the frequency of a larger bin made by combining say 10 bins is set totally by those nearest the center of the new bin.  This is the correct behavior I think, but can yield results that might surprise on first inspection.

So wanted to open this discussion, since `aggregate` only allows downsampling, should `resample_by_interpolation` only allow upsampling?   Should we also push their names slowly (without immediate breaking changes) to `upsample` and `downsample`?

**Comment by misi9170** [comment]
I'd support that. I think that aggregation is the appropriate way to downsample, and interpolating could lead to some odd results. Interpolating can probably also produce odd results when upsampling, but in that case it's a bit to be expected (since one is creating resolution that wasn't originally there)

**Comment by paulf81** [comment]
What if we add upsample and downsample for now as alias versions of the functions so we support either call, and then one day phase out the old names?

**Comment by misi9170** [comment]
Yeah, I would say we rename the functions to `upsample()` and `downsample()`, so that anyone coming to FLORIS from now sees that those is where the actual code lies, and then add new functions with the old names that are just aliases for the `upsample()` and `downsample()` methods... I think that's the way to do it?
